gprofiler/perf.py

```diff
@@ -2,77 +2,139 @@
 # Copyright (c) Granulate. All rights reserved.
 # Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
 #
-import concurrent.futures
 import os
-from tempfile import NamedTemporaryFile
+import signal
 from threading import Event
-from typing import Optional, Tuple
+from typing import List, Optional
 
 import psutil
 
+from gprofiler.exceptions import StopEventSetException
 from gprofiler.log import get_logger_adapter
 from gprofiler.merge import ProcessToStackSampleCounters, merge_global_perfs
 from gprofiler.profiler_base import ProfilerBase
-from gprofiler.utils import TEMPORARY_STORAGE_PATH, resource_path, run_process
+from gprofiler.utils import resource_path, run_process, start_process, wait_event, wait_for_file_by_prefix
 
 logger = get_logger_adapter(__name__)
 
-PERF_BUILDID_DIR = os.path.join(TEMPORARY_STORAGE_PATH, "perf-buildids")
+
+class PerfProcess:
```

Comment:
Looks like your can derive from ProfilerBase here and get away with much of the code

Reply:
Not really. It's not the same API. For example, its result isn't `ProcessToStackSampleCounters` but plain `perf script`.